### DISCUSSION TITLE
Using Frankenphp on AWS ECS

### DISCUSSION THREAD

**Discussion by magnetik** [php/frankenphp#790] [Q&A] [answered]
Hello,

We are currently running our apps on AWS ECS with php-hpm 8.2.

In order to have the task IAM role working, we need to get the `AWS_CONTAINER_CREDENTIALS_RELATIVE_URI` which is available only to the process started by ECS.

We were using this "hack" to get the AWS_CONTAINER_CREDENTIALS_RELATIVE_URI variable in our fpm env : we customize the fpm entrypoint and add this in the .sh: 

```shell
echo "env[AWS_CONTAINER_CREDENTIALS_RELATIVE_URI] = $(strings /proc/1/environ | grep AWS_CONTAINER_CREDENTIALS_RELATIVE_URI | cut -d = -f2 )" >> /usr/local/etc/php-fpm.d/www.conf

exec docker-php-entrypoint "$@"
```

(from https://github.com/aws/aws-sdk-php/issues/1971) 

How can I achieve something like that with Frankenphp? 

Thanks !

**Comment by dunglas** [comment, marked as answer]
As FrankenPHP is the process started, the env var should be available without having to do any hack.

**Reply by magnetik** [reply]
awesome ! Caddy does not run with muliple processes as fpm I guess. Thanks.
